### PR TITLE
fix(datetime-picker): update accessibility

### PR DESCRIPTION
**Pull request by kosol-komenek-rft**
## Description

Datetime picker change base class from ControlElement to FormFieldElement.

Fixes # (issue) https://jira.refinitiv.com/browse/ELF-2348

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/elements/src/datetime-picker/locales.ts [hunk 1]
```diff
@@ -21,20 +21,20 @@ import { Phrasebook } from '@refinitiv-ui/phrasebook';
 
 // match locales against date-fns
 // This will be used with resolveLocale function
-const globals = {};
+// for 5 langauge is set in Phrasebook de / en / ja / zh / zh-hant
+const globals = {
```
Comment (Theeraphat-Sorasetsakul) [inline packages/elements/src/datetime-picker/locales.ts]: Could you import this from phrasebook instead creating a new variable. I want to reduce redundant code.

File: packages/elements/src/datetime-picker/constants.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+enum DateTimePickerDataName {
+  from = 'from',
+  to = 'to'
+}
+
+export { DateTimePickerDataName };
```
Comment (Theeraphat-Sorasetsakul) [inline packages/elements/src/datetime-picker/constants.ts:6]: Could you re export this in element? So user can import this enum from element itself like slider do.

**Comment by Theeraphat-Sorasetsakul** [comment]
Curious on 2 things.
1. For now language feature of demo page doesn't work. Could you make demo page with translation like pagination demo? For example, To import `import '@refinitiv-ui/phrasebook/locale/ja/datetime-picker.js';`
2. Do we still need this? `protected t!: TranslateDirective;`

**Comment by kosol-komenek-rft** [comment]
> Curious on 2 things.
> 1. For now language feature of demo page doesn't work. Could you make demo page with translation like pagination demo? For example, To import `import '@refinitiv-ui/phrasebook/locale/ja/datetime-picker.js';`
> 2. Do we still need this? `protected t!: TranslateDirective;`

1 .we can add it . I add it in new commit
2.we can remove it . I remove it in new commit

**Review by Theeraphat-Sorasetsakul — DISMISSED** [review]
LGTM.

**Comment by bualoy-napat** [comment]
Verified!

**Review by wattachai-lseg — CHANGES_REQUESTED** [review]
There are a few concerns on EF doc: datetime pickers in all examples should be labelled. Plus, accessibility section needs an update. Please refer to https://ui.refinitiv.com/elements/text-field#accessibility.

File: documents/src/pages/elements/datetime-picker.md [hunk 3]
```diff
@@ -14,17 +14,28 @@ language_tabs: [javascript, typescript, css]
...
 [range][timepicker] {
   width: 400px;
 }
 ```
 ```html
 <section>
-  <ef-datetime-picker></ef-datetime-picker>
-  <ef-datetime-picker range duplex timepicker opened></ef-datetime-picker>
+  <div class="container">
+    <label for='datetime-picker'>Date of Birth</label>
```
Comment (wattachai-lseg) [inline documents/src/pages/elements/datetime-picker.md]: inconsistent quote usage. Let's stick with double quote.

File: documents/src/pages/elements/datetime-picker.md [hunk 4]
```diff
@@ -14,17 +14,28 @@ language_tabs: [javascript, typescript, css]
...
+    <ef-datetime-picker id='datetime-picker'></ef-datetime-picker>
```
Comment (wattachai-lseg) [inline documents/src/pages/elements/datetime-picker.md]: preferably, `id` should be more semantic e.g. `date-of-birth`.

File: packages/elements/src/datetime-picker/locales.ts [hunk 5]
```diff
@@ -14,27 +14,24 @@ import zhCN from 'date-fns/esm/locale/zh-CN/index.js';
...
+import translations from '@refinitiv-ui/phrasebook/locale/en/datetime-picker.js';
 
 // This file is a transition between using date-fns and Intl object to format dates
 // As of now, use Phraseboook to just resolve languages and locales
 // and match against the date-fns locales.
 
 // match locales against date-fns
 // This will be used with resolveLocale function
-const globals = {};
+// for 5 langauge is set in Phrasebook de / en / ja / zh / zh-hant
```
Comment (wattachai-lseg) [inline packages/elements/src/datetime-picker/locales.ts]:
```suggestion
// de, en, ja, zh & zh-hant have been defined in Phrasebook already.
```

File: packages/elements/src/datetime-picker/index.ts [hunk 6]
```diff
@@ -314,42 +332,12 @@ export class DatetimePicker extends ControlElement implements MultiValue {
...
   /**
    * Toggles the opened state of the list
    */
   @property({ type: Boolean, reflect: true })
   public opened = false;
 
-  /**
-   * Set error state
-   */
-  @property({ type: Boolean, reflect: true })
```
Comment (wattachai-lseg) [inline packages/elements/src/datetime-picker/index.ts:344]: Missing jsdoc for `error` & `warning` prop.

File: packages/elements/src/datetime-picker/index.ts [hunk 7]
```diff
@@ -1,10 +1,11 @@
 import inputFormat from 'date-fns/esm/format/index.js';
 import isValid from 'date-fns/esm/isValid/index.js';
 import inputParse from 'date-fns/esm/parse/index.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
```
Comment (wattachai-lseg) [inline packages/elements/src/datetime-picker/index.ts]: Let's move it down after `import {...} from '@refinitiv-ui/core';` as well.
```suggestion
import { ifDefined } from '@refinitiv-ui/core/directives/if-defined.js';
```

File: packages/elements/src/datetime-picker/index.ts [hunk 8]
```diff
@@ -87,6 +90,10 @@ const INPUT_TO_ID = 'input-to';
  * @attr {boolean} disabled - Set disabled state
  * @prop {boolean} [disabled=false] - Set disabled state
  *
```
Comment (wattachai-lseg) [inline packages/elements/src/datetime-picker/index.ts:81]: just 1 new line is enough.

File: packages/elements/src/datetime-picker/constants.ts [hunk 9]
```diff
@@ -0,0 +1,6 @@
+enum DateTimePickerDataName {
```
Comment (wattachai-lseg) [inline packages/elements/src/datetime-picker/constants.ts]: Let's name it meaningfully
```suggestion
enum DateTimePickerTranslationKeys {
```

File: packages/elements/src/datetime-picker/constants.ts [hunk 10]
```diff
@@ -0,0 +1,6 @@
...
+  from = 'from',
+  to = 'to'
```
Comment (wattachai-lseg) [inline packages/elements/src/datetime-picker/constants.ts]: Let's use `FROM` & `TO` so there is no need to convert their case later.

File: packages/elements/src/datetime-picker/index.ts [hunk 11]
```diff
@@ -537,6 +524,19 @@ export class DatetimePicker extends ControlElement implements MultiValue {
...
   }
 
+  /**
+   * Perform asynchronous update
+   * @returns promise
+   */
+  protected override async performUpdate(): Promise<void> {
+    [this.toAriaLabel, this.fromAriaLabel] = await Promise.all([
+      this.labelTPromise(DateTimePickerDataName.to.toUpperCase()),
+      this.labelTPromise(DateTimePickerDataName.from.toUpperCase())
```
Comment (wattachai-lseg) [inline packages/elements/src/datetime-picker/index.ts]: with update to `DateTimePickerDataName`, there should be need to call `toUpperCase()` now.

File: packages/elements/src/datetime-picker/index.ts [hunk 12]
```diff
@@ -519,6 +502,10 @@ export class DatetimePicker extends ControlElement implements MultiValue {
       this.syncInputValues();
     }
 
+    if (changedProperties.has('placeholder')) {
```
Comment (wattachai-lseg) [inline packages/elements/src/datetime-picker/index.ts:499]: let's change default value of `format` prop to `'dd-MMM-yyyy'` as well. Plus, add a comment saying that "`format` is the default value of `placeholder`"

File: documents/src/pages/elements/datetime-picker.md (hunk 3 above)
Comment (bualoy-napat) [reply documents/src/pages/elements/datetime-picker.md]: Fixed

File: documents/src/pages/elements/datetime-picker.md (hunk 4 above)
Comment (bualoy-napat) [reply documents/src/pages/elements/datetime-picker.md]: Fixed

File: packages/elements/src/datetime-picker/index.ts (hunk 6 above)
Comment (bualoy-napat) [reply packages/elements/src/datetime-picker/index.ts:344]: Fixed

File: packages/elements/src/datetime-picker/index.ts (hunk 8 above)
Comment (bualoy-napat) [reply packages/elements/src/datetime-picker/index.ts:81]: Fixed

File: packages/elements/src/datetime-picker/constants.ts (hunk 10 above)
Comment (bualoy-napat) [reply packages/elements/src/datetime-picker/constants.ts]: Fixed

File: packages/elements/src/datetime-picker/index.ts (hunk 11 above)
Comment (bualoy-napat) [reply packages/elements/src/datetime-picker/index.ts]: Fixed

File: packages/elements/src/datetime-picker/index.ts (hunk 12 above)
Comment (bualoy-napat) [reply packages/elements/src/datetime-picker/index.ts:499]: Fixed

**Review by dtanp-rft — DISMISSED** [review]
text box doesn't show correct locale. this should be fixed before release.

File: documents/src/pages/elements/datetime-picker.md [hunk 13]
```diff
@@ -649,7 +731,7 @@ ef-datetime-picker .custom-cell.selected {
 
 `ef-datetime-picker` provides input fields for users to enter date string values or date with time values. Users can open the popup with calendar element and use keyboard navigation to select the date from the UI.
 
-`ef-datetime-picker` has implemented keyboard navigation for users to navigate on the UI. You must ensure that the element has associated label by using `aria-label` or `aria-labelledby`.
+`ef-datetime-picker` has implemented keyboard navigation for users to navigate on the UI. You must ensure that the element has associated label by using `aria-label` or `aria-labelledby` or `for`.
```
Comment (wattachai-lseg) [inline documents/src/pages/elements/datetime-picker.md]:
```suggestion
`ef-datetime-picker` has implemented keyboard navigation for users to navigate on the UI. You must ensure that the element has associated label by using `aria-label` or `aria-labelledby` or `label[for="<element.id>"]`.
```

File: documents/src/pages/elements/datetime-picker.md [hunk 14]
```diff
@@ -662,5 +744,11 @@ ef-datetime-picker .custom-cell.selected {
   aria-labelledby="departure">
 </ef-datetime-picker>
 ```
+```html
+<label for="departure">Enter departure date</label>
```
Comment (wattachai-lseg) [inline documents/src/pages/elements/datetime-picker.md]: Let's apply this to the above 2 examples as well.
```suggestion
<label for="departure">Departure date</label>
```

File: packages/elements/src/datetime-picker/index.ts (hunk 12 above)
Comment (wattachai-lseg) [reply packages/elements/src/datetime-picker/index.ts:499]: still missing comment "`format` is the default value of `placeholder`"

File: documents/src/pages/elements/datetime-picker.md (hunk 13 above)
Comment (bualoy-napat) [reply documents/src/pages/elements/datetime-picker.md]: Fixed!

File: documents/src/pages/elements/datetime-picker.md (hunk 14 above)
Comment (bualoy-napat) [reply documents/src/pages/elements/datetime-picker.md]: Fixed!

**Review by wattachai-lseg — APPROVED** [review]

**Pull request merged**